### PR TITLE
MGMT-2463 - implement OCP Register day2 cluster

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

**Comment by avishayt** [comment]
Looks good to me so far.  Ronnie agree?

**Comment by yevgeny-shnaidman** [comment]
@avishayt  what about name?

File: cmd/main.go [hunk 1]
```diff
@@ -377,3 +393,13 @@ func uploadBaseISOWithLeader(uploadLeader leader.ElectorInterface, objectHandler
 		return err
 	})
 }
+
+func createOCPClusterWithLeader(uploadLeader leader.ElectorInterface, bm bminventory.OCPClusterAPI, log logrus.FieldLogger) error {
```
Comment (filanov) [inline cmd/main.go:397]: I think it belongs to main.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -3177,3 +3187,85 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
...
+		return err
+	}
+
+	return nil
+}
+
+func (b *bareMetalInventory) getApiVIPFromOCP(log logrus.FieldLogger) (string, error) {
+	configMap, err := b.k8sClient.GetConfigMap("kube-system", "cluster-config-v1")
+	if err != nil {
+		log.Errorf("Failed to get configmap cluster-config-v1 from namespace kube-system")
```
Comment (filanov) [inline internal/bminventory/inventory.go:3247]:
```suggestion
		log.WithError(err).Errorf("Failed to get configmap cluster-config-v1 from namespace kube-system")
```

Comment (filanov) [reply internal/bminventory/inventory.go:3247]: relevant to all the errors.

File: internal/bminventory/inventory_test.go [hunk 3]
```diff
@@ -3322,6 +3325,74 @@ var _ = Describe("proxySettingsForIgnition", func() {
...
+		bm = NewBareMetalInventory(db, getTestLog(), mockHostApi, mockClusterAPI, cfg, nil, nil, mockS3Client, mockMetric, getTestAuthHandler(), mockK8sClient)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	It("Register OCP cluster", func() {
+		configMap := v1.ConfigMap{}
+		configMap.Data = make(map[string]string)
```
Comment (filanov) [inline internal/bminventory/inventory_test.go:3360]: common content should be in before each or in a function

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory_test.go]: fixed

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go]: fixed

File: cmd/main.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply cmd/main.go]: fixed

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -3186,3 +3196,85 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
...
+	err = validations.ValidateClusterNameFormat(clusterName)
+	if err != nil {
+		log.Errorf("Failed to validate cluster name: %s", clusterName)
+		return err
+	}
+
+	// Persist worker-ignition to s3 for cluster
+	err = b.createAndUploadNodeIgnition(ctx, &id, apiVIP)
+	if err != nil {
+		log.Errorf("Failed to create and upload worker ignition for cluster %s", id)
```
Comment (filanov) [inline internal/bminventory/inventory.go:3239]:
```suggestion
		log.WithError(err).Errorf("Failed to create and upload worker ignition for cluster %s", id)
```
relevant to all the logs.

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -3186,3 +3196,85 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
...
+		log.WithError(err).Errorf("Failed to get configmap cluster-config-v1 from namespace kube-system")
+		return "", err
+	}
+	configStruct := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(configMap.Data["install-config"]), configStruct)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal confimap cluster-config-v1 data: <%s>", configMap.Data["install-config"])
+		return "", err
+	}
+	platform := configStruct["platform"].(map[interface{}]interface{})
```
Comment (filanov) [inline internal/bminventory/inventory.go:3265]: Do you need to check that it's not nil? 
you will have panic if something is not set correctly.

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go]: fixed

File: internal/bminventory/inventory.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go]: fixed

File: internal/bminventory/inventory.go [hunk 6]
```diff
@@ -3186,3 +3196,100 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
...
+	return apiVip, nil
+}
+
+func (b *bareMetalInventory) getOpenshiftVersionFromOCP(log logrus.FieldLogger) (string, error) {
+	clusterVersion, err := b.k8sClient.GetClusterVersion("version")
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get cluster version from OCP")
+		return "", err
+	}
+	openshiftVersion := clusterVersion.Status.Desired.Version
```
Comment (filanov) [inline internal/bminventory/inventory.go:3292]: Ca one of the those fields be nil?

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:3292]: no,they are all structs, not pointers

**Comment by filanov** [comment]
I don't like the lack of tests

**Review by filanov — APPROVED** [review]

**Pull request merged**